src/formula/callable_objects.cpp

```diff
@@ -255,7 +256,20 @@ variant unit_callable::get_value(const std::string& key) const
 	} else if(key == "max_attacks") {
 		return variant(u_.max_attacks());
 	} else if(key == "traits") {
+		deprecated_message("traits", DEP_LEVEL::INDEFINITE, "1.17.20", "Use traits_ids instead");
```

Comment:
This is "the earliest version it could possibly be removed in", not "the version it was deprecated in". So, "1.19" is the obvious choice, though since it's an indefinite deprecation you could also make it a bit later, like "1.21".

Reply:
removed with ids suffix, impossible to deprecate if not alternative and 'modifications.advancements" is too complicated to put in formula

Reply:
I never asked for it to be put in formula.

Reply:
i already removed this commit